pvlib/temperature.py

```diff
@@ -423,3 +424,199 @@ def faiman(poa_global, temp_air, wind_speed=1.0, u0=25.0, u1=6.84):
+    # combine free and forced components
+    hconv = (hfree**3 + hforce**3)**(1/3)
+    return hconv
+
+
+def fuentes(poa_global, temp_air, wind_speed, noct_installed, module_height=5,
+            wind_height=9.144, emissivity=0.84, absorption=0.83,
+            hydraulic_diameter=0.5):
+    """
+    Calculate cell or module temperature using the Fuentes model.
+
+    The Fuentes model is a first-principles heat transfer energy balance
+    model [1]_ that is used in PVWatts for cell temperature modeling [2]_.
+
+    Parameters
+    ----------
+    poa_global : pandas Series
+        Total incident irradiance [W/m^2]
+
+    temp_air : pandas Series
+        Ambient dry bulb temperature [C]
+
+    wind_speed : pandas Series
+        Wind speed [m/s]
+
+    noct_installed : float
+        The "installed" nominal operating cell temperature as defined in [1]_.
+        PVWatts assumes this value to be 45 C for rack-mounted arrays and
+        49 C for roof mount systems with restricted air flow around the
+        module.  [C]
+
+    module_height : float, default 5.0
+        The height above ground of the module above the ground. The PVWatts
+        default is 5.0 [m]
+
+    wind_height : float, default 9.144
+        The height above ground at which ``wind_speed`` is measured. The
+        PVWatts defauls is 9.144 [m]
+
+    emissivity : float, default 0.84
+        The effectiveness of the module at radiating thermal energy. [unitless]
+
+    absorption : float, default 0.83
+        The fraction of incident irradiance that is converted to thermal
+        energy in the module. [unitless]
+
+    hydraulic_diameter : float, default 0.5
+        The hydraulic diameter of the module. The default value of 0.5 is
+        provided in [1]_ for a module with dimensions 0.3m x 1.2m.  [m]
+
+    Returns
+    -------
+    temperature_cell : pandas Series
+        The modeled cell temperature [C]
+
+    Notes
+    -----
+    This function returns slightly different values from PVWatts at night
+    and just after dawn. This is because the SAM SSC assumes that module
+    temperature equals ambient temperature when irradiance is zero so it can
+    skip the heat balance calculation at night.
+
+    References
+    ----------
+    .. [1] Fuentes, M. K., 1987, "A Simplifed Thermal Model for Flat-Plate
+           Photovoltaic Arrays", SAND85-0330, Sandia National Laboratories,
+           Albuquerque NM.
+           http://prod.sandia.gov/techlib/access-control.cgi/1985/850330.pdf
+    .. [2] Dobos, A. P., 2014, "PVWatts Version 5 Manual", NREL/TP-6A20-62641,
+           National Renewable Energy Laboratory, Golden CO.
+           doi:10.2172/1158421.
+    """
+    # ported from the FORTRAN77 code provided in Appendix A of Fuentes 1987;
+    # nearly all variable names are kept the same for ease of comparison.
+
+    boltz = 5.669e-8
+    emiss = emissivity
+    absorp = absorption
+    xlen = hydraulic_diameter
+    cap0 = 11000
```

Comment:
`cap0` accounts for module area and mass, correct?

Reply:
My understanding is that `cap0` is the product `m c` where `m` is the mass per unit area [kg/m^2] and `c` is the specific heat of the whole module [J / (kg K)], so `cap0` has units [J / (m^2 K)].  The 11,000 comes from Table 3 on page 8.

Reply:
```suggestion
    cap0 = 11000  # units of [J / (m^2 K)], equal to mass per unit area times specific heat of the module.
```